cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/validator/PreCompileValidatorImpl.java

```diff
@@ -203,6 +206,38 @@ public void validateResultsSection(
         }
     }
 
+    @Override
+    public void validateNoDuplicateInputs(List<Input> inputs, Input element) {
```

Comment:
these four methods look similar, maybe they can be replaced by a generic method with a List<InOutParam> and an InOutParam as parameters